test/filters.js

```diff
@@ -707,6 +713,93 @@ describe('filters', function () {
     })
   })
 
+  describe("toCurrency", function(){
+    let errorMessage = "invalid currency value or type";
+
+    it('should return {value: 8000, type: "INR"}"', () => {
+      const dst = { value: 8000, type: "INR" };
+      return test('{{ 8000 | toCurrency: "INR" }}', JSON.stringify(dst));
+    });
+
+    it('should return {value: 100, type: "USD"} for params 100 and "USD"', () => {
+      const dst = { value: 100, type: "USD" };
+      return test('{{ 100 | toCurrency: "USD" }}', JSON.stringify(dst));
+    });
+
+    it('should return {value: 100, type: "USD"} for params 100 and "USD"', () => {
+      const dst = { value: 150.1, type: "USD" };
+      return test('{{ 150.1 | toCurrency: "USD" }}', JSON.stringify(dst));
+    });
+
+    it("should throw error when currency value is 0", () =>
+      checkForError('{{ 0 | toCurrency: "USD" }}', errorMessage));
+
+    it("should throw error when currency value is negative number", () =>
```

Comment:
Should liquid care if the currency value is negative? For example if we chain few filters together, we might break a valid chain because of this restriction.

`{% assign nine_hundred_dollar = 10 | minus: 110 | toCurrency: "USD" | plus: thousand_dollar %}`

Assuming `thousand_dollar` is a valid currency object, the above assignment should assign 900$ to the `nine_hundred_dollar` variable.

But if we don't allow negative values in `toCurrency`, the computation will fail. Similarly for `toDuration`.

@krishi-vb @anirudhvarma12

Reply:
Makes sense, have removed this condition from filter code and test.